manifests/config.pp

```diff
@@ -0,0 +1,169 @@
+# == Class nzbget::config
+#
+# This class is called from nzbget for service config.
+#
+
+class nzbget::config {
+  if ($::nzbget::manage_service_dirs[0]) {
+    file { $::nzbget::managed_service_dirs:
+      ensure => directory,
+      owner  => $::nzbget::user,
+      group  => $::nzbget::group,
+      mode   => '0750',
+    }
+  }
+
+  if ($::nzbget::manage_data_dirs[0]) {
+    file { $::nzbget::managed_data_dirs:
+      ensure => directory,
+      owner  => $::nzbget::user,
+      group  => $::nzbget::group,
+      mode   => '0750',
+    }
+  }
+
+  if ($::nzbget::manage_certs) {
+    if ($::nzbget::secure_cert) {
+      file { $::nzbget::secure_cert:
+        ensure => present,
+        owner  => $::nzbget::user,
+        group  => $::nzbget::group,
+        mode   => '0400',
+        source => "puppet:///modules/nzbget/${nzbget::secure_cert_file}",
+      }
+    }
+
+    if ($::nzbget::secure_key) {
+      file { $::nzbget::secure_key:
+        ensure => present,
+        owner  => $::nzbget::user,
+        group  => $::nzbget::group,
+        mode   => '0400',
+        source => "puppet:///modules/nzbget/${nzbget::secure_key_file}",
```

Comment:
This way of getting keys is shit. It should be done by passing in a parameter, the key can then be stored in hiera and the module can be re-used without modification.

Reply:
Same goes for similar instances above and below